### PR TITLE
[Build Scripts] Enable OS X build when Xcode is located in a path con…

### PR DESCRIPTION
**Pull request by k8stone**
…taining spaces

Applied a combination of explicitly quoted paths and bash's string escaping facilities to allow a build to complete successfully.

### CODE REVIEW

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -46,12 +46,12 @@ function(_add_variant_c_compile_link_flags
     "-target" "${SWIFT_SDK_${sdk}_ARCH_${arch}_TRIPLE}")
 
   list(APPEND result
-    "-isysroot" "${SWIFT_SDK_${sdk}_PATH}")
+    "-isysroot" "\"${SWIFT_SDK_${sdk}_PATH}\"")
```
Comment (jrose-apple) [inline cmake/modules/AddSwift.cmake]: The CMake changes here seem odd to me. This is a list and these paths are already quoted, which means at this point in time CMake knows what one element consists of. Can we accomplish this using the `VERBATIM` setting for all `add_custom_command_target` calls instead?

(I'd be fine with forcing this setting in `add_custom_command_target` itself, if it doesn't break anything.)

File: utils/build-script-impl [hunk 2]
```diff
@@ -241,11 +241,11 @@ function set_deployment_target_based_options() {
                     cmark_cmake_options=(
                         -DCMAKE_C_FLAGS="$(cmark_c_flags $deployment_target)"
                         -DCMAKE_CXX_FLAGS="$(cmark_c_flags $deployment_target)"
-                        -DCMAKE_OSX_SYSROOT:PATH="$(xcrun --sdk $xcrun_sdk_name --show-sdk-path)"
+                        -DCMAKE_OSX_SYSROOT:PATH=$(printf %q "$(xcrun --sdk $xcrun_sdk_name --show-sdk-path)")
```
Comment (gribozavr) [inline utils/build-script-impl]: These paths are also already quoted, I feel like the issue is in some other place.

**Comment by gribozavr** [comment]
This code is the issue, it should be fixed instead:

```
  # Convert variables to space-separated strings.
  string(REPLACE ";" " " c_compile_flags "${c_compile_flags}")
  string(REPLACE ";" " " link_flags "${link_flags}")
```

**Comment by gribozavr** [comment]
https://github.com/apple/swift/commit/ba6643c7a4b851a131cd84616069f932747b5a7b is the first step.  The build passes, tests still have issues.

File: utils/build-script-impl (hunk 2 above)
Comment (k8stone) [inline utils/build-script-impl]: While the paths are quoted here, they are subsequently used in other locations where they are inconsistently quoted.  Escaping whitespace in the path itself via printf %q resolved the related build issues I observed.

File: cmake/modules/AddSwift.cmake (hunk 1 above)
Comment (k8stone) [inline cmake/modules/AddSwift.cmake]: I'm not deeply attached to any particular strategy, just trying to unblock usage with one of my common configurations.  Alternatives that achieve the same end result are equally welcome.  There may be unnecessary changes here as I had to iterate a bit to get the build process to work at all when my Xcode path contains a space.  I can try to isolate which changes are strictly required and which are just aggressively paranoid.

Comment (gribozavr) [inline cmake/modules/AddSwift.cmake]: I have landed a patch that allowed me to bulid with an Xcode with spaces in the path.  The tests still had issues.  Could you pick up the work from there?  The next change should be in lit.cfg.

**Comment by k8stone** [comment]
I'm content that Dmitri's approach above has addressed the issues this pull request was intended to solve.  Thanks!  Solving the remaining test issues would definitely be appreciated.

**Pull request closed without merging**